### PR TITLE
fix: support dynamically added components #8016

### LINKED ISSUE #8016: [Magellan] active class not being applied when content is loaded dynamically
**How can we reproduce this bug**
1. Create a magellan nav with sticky
2. Populate content dynamically (angular)
3. Initialize foundation js after content is loaded

**What did you expect to happen** 
Magellan should apply an 'active' class when the element is in view

**What happened instead** 
No active classes are applied to the menu. 

I'm using F6 in an angular application and i'm trying to use ng-repeat with the requisite magellan markup. I'm initializing the foundation js after content is loaded. plugins (sticky, accordion, dropdown) seem to work, just not magellan.

Codepen: http://codepen.io/npapazian/pen/MKVGrr

### PR DESCRIPTION
**Pull request by DanielRuf**
We have to check for `document.readyState === "complete"` when the window and content is already loaded and more content is loaded afterwards that is needed for the magellan and sticky solution.

### CODE REVIEW

**Comment by DanielRuf** [comment]
Working codepen: https://codepen.io/DanielRuf/pen/LQOWmE

**Comment by ncoden** [comment]
I just found the same issue with `Magellan`. There is probably others issues like this across all components. :/

**Comment by DanielRuf** [comment]
Right, we may have to add this to some more components.

**Comment by DanielRuf** [comment]
I'll check the other components, prepare an issue and collect them there and link the PRs then (I would rather go step for step or component for component to have better and more atomar changes which is easier for bisecting and regression testing imo).

**Comment by ncoden** [comment]
I was going to take care of this and open a PR but ok, have fun ;)

**Comment by DanielRuf** [comment]
> keeping everything in one PR is easier to manage.
Agreed. I'll prepare the other commits then. If you want you can also push to my branch. Or do we already have some PRs for the other components?

**Comment by ncoden** [comment]
>  Or do we already have some PRs for the other components?

I don't think. There was PRs before that resolved some cases manually (and sometimes badly), but everything is merged by now.

**Comment by DanielRuf** [comment]
Also see https://github.com/zurb/foundation-sites/issues/9047

**Comment by ncoden** [comment]
Thanks.

**Comment by DanielRuf** [comment]
Pushed a new commit. Did not find more occurences where we wait for the load event.

**Comment by ncoden** [comment]
@DanielRuf Seen. LGTM but require some cleaning as said at https://github.com/zurb/foundation-sites/pull/10988#discussion_r171405136, maybe in an other PR (that or in that PR but with a squash after). Do you want to take care of this or should I ?

**Comment by DanielRuf** [comment]
> LGTM but require some cleaning as said at #10988 (comment), maybe in an other PR (that or in that PR but with a squash after). Do you want to take care of this or should I ?

Would be glad if I couls get some assistance here so feel free to prepare a PR with the needed changes to make them all consistent then.

File: js/foundation.magellan.js [hunk 1]
```diff
@@ -83,15 +83,25 @@ class Magellan extends Plugin {
...
+      $(window).one('load', function(){
+        if(_this.options.deepLinking){
+          if(location.hash){
+            _this.scrollToLoc(location.hash);
+          }
+        }
+        _this.calcPoints();
+        _this._updateActive();
+      });
+    }
```
Comment (ncoden) [inline js/foundation.magellan.js]: > When  a Magellan component is dynamically created, the window DO NOT scroll  to the target corresponding to the location hash as the magellan did not  created the target and should appear without effects on it after the  page already loaded.
>
> -- https://github.com/zurb/foundation-sites/pull/11077#issue-175772567

**Review by ncoden — CHANGES_REQUESTED** [review]

**Comment by ncoden** [comment]
Closing in favor of https://github.com/zurb/foundation-sites/pull/11077

**Pull request closed without merging**